src/runtime/virtcontainers/fs_share_linux.go

```diff
@@ -666,12 +696,14 @@ func (f *FilesystemShare) StartFileEventWatcher(ctx context.Context) error {
 					f.Logger().Infof("StartFileEventWatcher: dataDir (%s)", dataDir)
 					// eg. dataDir = /var/lib/kubelet/pods/b44e3261-7cf0-48d3-83b4-6094bba95dc8/volumes/kubernetes.io~configmap/foo/..data
 
-					destination := f.srcDstMap[dataDir]
-					f.Logger().Infof("StartFileEventWatcher: Copy file from src (%s) to dst (%s)", dataDir, destination)
-					err := f.copyFilesFromDataDir(dataDir, destination)
-					if err != nil {
-						f.Logger().Infof("StartFileEventWatcher: got an error (%v) when copying file from src (%s) to dst (%s)", err, dataDir, destination)
-						return err
+					// Handle different destination for the same source
+					for _, destination := range f.srcDstMap[dataDir] {
```

Comment:
What prevents a concurrent writer to conflict with this read?

Unless there's some synchronization here I'm missing, you need to acquire `srcDstMapLock` here too to prevent modifications to `srcDstMap` while you're iterating over it.

Reply:
You are right. Will make the changes.